processing/src/main/java/org/apache/druid/segment/AutoTypeColumnIndexer.java

```diff
@@ -405,8 +405,8 @@ public ColumnType getLogicalType()
       return ColumnType.NESTED_DATA;
     }
     if (isConstant && constantValue == null) {
-      // we didn't see anything, so we can be anything, so why not a string?
-      return ColumnType.STRING;
+      // we didn't see anything, so we can be anything, so be long as it is the most restrictive type
+      return ColumnType.LONG;
```

Comment:
If this column is seen previously, should it just return that type?

Reply:
the issue here is that there have been no values other than nulls so far, so we can't know a type yet

Reply:
In the scenario when the ingestion spec doesn't have any custom dimensions, new incremental index is initialized with columns from previous index https://github.com/apache/druid/blob/master/server/src/main/java/org/apache/druid/segment/realtime/sink/Sink.java#L379, in that case it would probably be better to return the previous seen type?